### PR TITLE
Add Amplitude events for opening and closing FAB and LearningGoals

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Working off [Tess's doc](https://docs.google.com/document/d/1YQ2rgvOYRpn4I1psxLHnjlIsuDe_F9rzkSg1Tj9wb3M/edit), adding a few early Amplitude events for components I created in previous PRs. 

Out of that docs, I think this covers `teacher_open_ai_ta` and `teacher_expand_rubric`. It partially covers `teacher_abandon_rubric` with the close/collapse events.

Example logging: 
```
[AMPLITUDE ANALYTICS EVENT]: Rubric Learning Goal Expanded. Payload: {"payload":{"learningGoal":"Program Development"}}
```

I haven't gotten everything hooked up yet so there wasn't an easy way to test the FAB events. I'll make sure to test them once everything comes together.

### CODE REVIEW

File: apps/src/templates/rubrics/LearningGoal.jsx [hunk 1]
```diff
@@ -2,27 +2,39 @@ import React, {useState} from 'react';
...
   BodyFourText,
 } from '@cdo/apps/componentLibrary/typography';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
 
 export default function LearningGoal({
   learningGoal,
   teacherHasEnabledAi,
   canProvideFeedback,
+  reportingData,
```
Comment (bethanyaconnor) [inline apps/src/templates/rubrics/LearningGoal.jsx:21]: This is a pattern I'm trying out. We often want context information for these events that we don't need for the actual rendering of the component. Definitely open to feedback on this!

File: apps/src/templates/rubrics/LearningGoal.jsx [hunk 2]
```diff
@@ -2,27 +2,39 @@ import React, {useState} from 'react';
...
   const aiEnabled = learningGoal.aiEnabled && teacherHasEnabledAi;
 
+  const handleClick = () => {
+    const eventName = isOpen
+      ? EVENTS.RUBRIC_LEARNING_GOAL_COLLAPSED_EVENT
+      : EVENTS.RUBRIC_LEARNING_GOAL_EXPANDED_EVENT;
+    analyticsReporter.sendEvent(eventName, {
+      ...(reportingData || {}),
+      learningGoalKey: learningGoal.key,
+      learningGoal: learningGoal.learningGoal,
```
Comment (bethanyaconnor) [inline apps/src/templates/rubrics/LearningGoal.jsx:34]: Adding both the key and the text here as, once we translate this feature, `learningGoal` will likely be impossible to aggregate across languages.

File: apps/src/templates/rubrics/LearningGoal.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/templates/rubrics/LearningGoal.jsx:21]: Tagging @hannahbergam in case you have any thoughts on this!

**Review by kobryan0619 — APPROVED** [review]
This looks good to me!  Also interested in Hannah's thoughts about the way to pass the event data.  

One other question, is it common/expected to write tests for amplitude events?  I haven't written any amplitude events yet myself, so this is more just for my own understanding.

File: apps/src/templates/rubrics/LearningGoal.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/src/templates/rubrics/LearningGoal.jsx:21]: Oooo I like this pattern. Makes me think we could refactor a few other places to match!

**Review by hannahbergam — APPROVED** [review]
LGTM! Also to answer your question @kobryan0619 , I think we've been only writing test cases if we aren't able to test and verify them locally, so makes sense here! Otherwise, I often just make sure all the test call permutations look right locally and verify with the PM that they're showing up as expected.

**Pull request merged**